equational_theories/EquationsCommand.lean

```diff
@@ -9,12 +9,10 @@ def mkNatMagmaLaw (declName : Name) : ImportM NatMagmaLaw := do
   let { env, opts, .. } ← read
   IO.ofExcept <| unsafe env.evalConstCheck NatMagmaLaw opts ``NatMagmaLaw declName
 
-initialize magmaLawExt : PersistentEnvExtension Name (Name × NatMagmaLaw) (Array (Name × NatMagmaLaw)) ←
-  registerPersistentEnvExtension {
-    mkInitial := pure .empty
-    addImportedFn := Array.concatMapM <| Array.mapM <| fun n ↦ do return (n, ← mkNatMagmaLaw n)
```

Comment:
> For most extensions, α and β coincide. `α` and ‵β` do not coincide for extensions where the data
used to update the state contains elements which cannot be stored in files (for example, closures).

https://github.com/leanprover/lean4/blob/d10d41bc07942ca6d8f3081a637045321db15c5a/src/Lean/Environment.lean#L444-L445


I think this means that extension ought to have been declared as having type
```
PersistentEnvExtension (Name × NatMagmaLaw) (Name × NatMagmaLaw) (Array (Name × NatMagmaLaw))
```

in which case `addImportedFun` could already have the `NatMagmaLaw` from its imports, and would not need to recreate it with `mkNatMagmaLaw`.

Reply:
In any case, using `SimplePersistentEnvironmentExtension` avoids this confusing interface and seems to perform well.

Reply:
I believe that when a `PersistentEnvironmentExtension` is stored in an `.olean` file, only the name of the declaration gets stored (since the full datastructure is difficult to serialize), and when it is imported into a file, the `mkNatMagmaLaw` function is called to construct the full state of the environment extension out of the names. It's a code pattern that I've observed in other environment extensions defined in Lean core, which is why I decided to include it in this definition.

As I type this, I realize that this may be a reason for the slow-down -- every file that imports `AllEquations.lean` must run `mkNatMagmaLaw` on every single law defined each time, which could be taking a lot of time and clogging memory.

Reply:
And by the way, thank you so much for helping fix the issue with the environment extension.

Reply:
Would it make sense to only store `Array Name` in the extension, and run `mkNatMagma` only when actually accessing a value from it? (I didn’t look closely what this extension is used for.)

Reply:
Or use the existing `Lean.TagAttribute`?

Reply:
Yes, that sounds reasonable to me.

Reply:
I've added a PR for this in @dwrensha's repository. Thanks for the suggestion.